parse/EffectPythonParser.cpp

```diff
@@ -287,6 +287,26 @@ namespace {
             auto target = py::extract<condition_wrapper>(kw["target"])();
             return effect_wrapper(std::make_shared<Effect::MoveTowards>(std::move(speed),
                 ValueRef::CloneUnique(target.condition)));
+        } else if (kw.has_key("x") && kw.has_key("y")) {
+            std::unique_ptr<ValueRef::ValueRef<double>> x;
+            auto x_args = boost::python::extract<value_ref_wrapper<double>>(kw["x"]);
+            if (x_args.check()) {
+                x = ValueRef::CloneUnique(x_args().value_ref);
+            } else {
+                x = std::make_unique<ValueRef::Constant<double>>(boost::python::extract<double>(kw["x"])());
+            }
+
+            std::unique_ptr<ValueRef::ValueRef<double>> y;
```

Comment:
Can we have some helpers to avoid code duplication?

std::unique_ptr<ValueRef::ValueRef<double>> y = get_double_argument(kw, "y")

Reply:
Sounds reasonable